node/events.d.ts

```diff
@@ -0,0 +1,664 @@
+      emitter: DOMEventTarget,
+      eventName: string,
+      options?: StaticEventEmitterOptions,
+    ): Promise<any[]>;
+    /**
+     * ```js
+     * const { on, EventEmitter } = require('events');
+     *
+     * (async () => {
+     *   const ee = new EventEmitter();
+     *
+     *   // Emit later on
+     *   process.nextTick(() => {
+     *     ee.emit('foo', 'bar');
+     *     ee.emit('foo', 42);
+     *   });
+     *
+     *   for await (const event of on(ee, 'foo')) {
+     *     // The execution of this inner block is synchronous and it
+     *     // processes one event at a time (even with await). Do not use
+     *     // if concurrent execution is required.
+     *     console.log(event); // prints ['bar'] [42]
+     *   }
+     *   // Unreachable here
+     * })();
+     * ```
+     *
+     * Returns an `AsyncIterator` that iterates `eventName` events. It will throw
+     * if the `EventEmitter` emits `'error'`. It removes all listeners when
+     * exiting the loop. The `value` returned by each iteration is an array
+     * composed of the emitted event arguments.
+     *
+     * An `AbortSignal` can be used to cancel waiting on events:
+     *
+     * ```js
+     * const { on, EventEmitter } = require('events');
+     * const ac = new AbortController();
+     *
+     * (async () => {
+     *   const ee = new EventEmitter();
+     *
+     *   // Emit later on
+     *   process.nextTick(() => {
+     *     ee.emit('foo', 'bar');
+     *     ee.emit('foo', 42);
+     *   });
+     *
+     *   for await (const event of on(ee, 'foo', { signal: ac.signal })) {
+     *     // The execution of this inner block is synchronous and it
+     *     // processes one event at a time (even with await). Do not use
+     *     // if concurrent execution is required.
+     *     console.log(event); // prints ['bar'] [42]
+     *   }
+     *   // Unreachable here
+     * })();
+     *
+     * process.nextTick(() => ac.abort());
+     * ```
+     * @since v13.6.0, v12.16.0
+     * @param eventName The name of the event being listened for
+     * @return that iterates `eventName` events emitted by the `emitter`
+     */
+    static on(
+      emitter: NodeJS.EventEmitter,
+      eventName: string,
+      options?: StaticEventEmitterOptions,
+    ): AsyncIterableIterator<any>;
+    /**
+     * A class method that returns the number of listeners for the given `eventName`registered on the given `emitter`.
+     *
+     * ```js
+     * const { EventEmitter, listenerCount } = require('events');
+     * const myEmitter = new EventEmitter();
+     * myEmitter.on('event', () => {});
+     * myEmitter.on('event', () => {});
+     * console.log(listenerCount(myEmitter, 'event'));
+     * // Prints: 2
+     * ```
+     * @since v0.9.12
+     * @deprecated Since v3.2.0 - Use `listenerCount` instead.
+     * @param emitter The emitter to query
+     * @param eventName The event name
+     */
+    static listenerCount(
+      emitter: NodeJS.EventEmitter,
```

Comment:
This could be avoided by merging the NodeJS.EventEmitter declaration into the EventEmitter one